backend/src/main/java/de/neuefische/booklibrary/backend/security/AppUserService.java

```diff
@@ -0,0 +1,50 @@
+package de.neuefische.booklibrary.backend.security;
+
+import de.neuefische.booklibrary.backend.book.UserNotExistsByUsernameException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+
+@Service
+@RequiredArgsConstructor
+public class AppUserService {
+
+    private final AppUserRepository appUserRepository;
+
+    public AppUser findByUsername(String username) {
+        return appUserRepository.findByUsername(username).orElseThrow(() -> new UserNotExistsByUsernameException("No user found with this username"));
```

Comment:
In diesem Fall würde ich ausnahmsweise für ein `throws` für die UncheckedException plädieren - um dem Aufrufer klar zu machen, dass es hier entweder den User oder eine Exception als Ergebnis gibt.
```suggestion
    public AppUser findByUsername(String username) throws UserNotExistsByUsernameException {
        return appUserRepository.findByUsername(username).orElseThrow(() -> new UserNotExistsByUsernameException("No user found with this username"));
```

Reply:
macht Sinn :)